src/app/dynamic/enterprise/metering/old-reports/list/component.ts

```diff
@@ -0,0 +1,132 @@
+import {MatPaginator} from '@angular/material/paginator';
+import {Report} from '@shared/entity/metering';
+import {MeteringService} from '@app/dynamic/enterprise/metering/service/metering';
+import {filter, switchMap, take, takeUntil} from 'rxjs/operators';
+import {MatDialog, MatDialogConfig} from '@angular/material/dialog';
+import {ConfirmationDialogComponent} from '@app/shared/components/confirmation-dialog/component';
+import {NotificationService} from '@app/core/services/notification';
+import {Subject} from 'rxjs';
+
+enum Column {
+  name = 'name',
+  size = 'size',
+  lastModified = 'lastModified',
+  actions = 'actions',
+}
+
+@Component({
+  selector: 'km-old-metering-reports-list',
+  templateUrl: './template.html',
+})
+export class MeteringOldReportListComponent implements OnInit {
+  private readonly _unsubscribe = new Subject<void>();
+  @Input() reports: Report[];
+
+  readonly column = Column;
+  readonly displayedColumns: string[] = Object.values(Column);
+
+  private _reportsInProgress = new Set<string>([]);
+
+  dataSource = new MatTableDataSource<Report>();
+  @ViewChild(MatSort, {static: true}) sort: MatSort;
+  @ViewChild(MatPaginator, {static: true}) paginator: MatPaginator;
+
+  constructor(
+    private readonly _meteringService: MeteringService,
+    private readonly _matDialog: MatDialog,
+    private readonly _notificationService: NotificationService,
+    @Inject(DOCUMENT) private readonly _document: Document
+  ) {}
+
+  ngOnInit() {
+    this.dataSource.data = this.reports;
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.sort.active = 'lastModified';
+    this.sort.direction = 'desc';
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  download(reportName: string): void {
+    this._reportsInProgress.add(reportName);
+
+    this._meteringService
+      .reportDownload(reportName)
+      .pipe(takeUntil(this._unsubscribe))
+      .subscribe({
+        next: url => {
+          this._reportsInProgress.delete(reportName);
+          this._document.defaultView?.open(url, '_blank');
+        },
+        complete: () => this._reportsInProgress.delete(reportName),
+        error: _ => this._reportsInProgress.delete(reportName),
+      });
+  }
+
+  remove(reportName: string): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Delete Metering Report',
+        message: `Do you want to delete <b>${reportName}</b> report permanently?`,
+        confirmLabel: 'Delete',
+        warning: 'This change is permanent.',
+      },
+    };
+    this._matDialog
+      .open(ConfirmationDialogComponent, dialogConfig)
+      .afterClosed()
+      .pipe(filter(isConfirmed => isConfirmed))
+      .pipe(
+        switchMap(_ => {
+          this._reportsInProgress.add(reportName);
+          this.dataSource.data = this.reports.filter(report => report.name !== reportName);
```

Comment:
Shouldn't these 2 lines be inside `subscribe` so that view is updated only when the request finishes?